frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -257,9 +257,16 @@ const composeAndPushNotificationMessages = (
             getSelectedRole()?.role_name
         );
     }
-    if (errorIdList.length > 0) {
+
+    if(errorIdList.errorMsg) {
+        setNotificationMsg(
+        Severity.error,
+        `An error has occured. ${errorIdList.errorMsg}`
+        );
+    };
+    if (errorIdList.forestClienNumber.length > 0) {
```

Comment:
The above 2 "if" statements (line 261 and line 267) look suspicious at the intention of the `const errorForestClientIdList = {errorMsg: '', forestClienNumber: [] as string[]};` usage.

It simply states like this:
```
if(errorIdList.errorMsg) {
   // call function A to handle message
}
if (errorIdList.forestClienNumber.length > 0) {
   //  call function B to handle message - with ids.
}
```
So developer may wonder are these two properties "**errorMsg**" and "**forestClienNumber**" mutually exclusive in ` {errorMsg: '', forestClienNumber: [] as string[]}`?
Is there a third condtion when both **errorMsg** and **forestClienNumber** exist? And how the code is going to react with that? From **above**, does it intend to produce "**two**" messages?
Or if it cannot happen the same time, why not just use "**if(){ // do this... } ... else{ // do that... }**" to be clear?

More over, as mentioned above why the "**error**" variable definition/structure is different than "**success/warning**" when the name of "**Notification**" function sounds generic enough? I feel adding/displaying notifications are getting confusing and might need some thought later.

Reply:
I'll see what can be done to make it better, thanks!